### ISSUE TITLE
Session Recording Playback fails based on terminal resolution size when recording vim

### ISSUE THREAD

**Issue by pschisa** [gravitational/teleport#35756]
Expected behavior:
Teleport will present a auditable session recording with the `tsh play` in a terminal when vim is used during the session

Current behavior:
If the terminal is smaller than the one used to make the recording and vim is used, the recording becomes mangled upon replay during the vim session.

https://github.com/gravitational/teleport/assets/75806143/8d988462-0d5d-4c0f-b21b-76c23b0f7d84

Bug details:
- Teleport version: 
Teleport v14.1.3 git:v14.1.3-0-g748fa4e go1.21.4
Proxy version: 14.2.0

**Comment by pschisa** [comment]
Workaround is to utilize a larger terminal or the web UI for session recording replay

**Comment by zmb3** [comment]
I think Teleport is doing the right thing here. We capture the size of the terminal in the session recording (both at startup and any time it changes thereafter), and during playback we emit the ANSI escape codes to resize the window.

It's up the the user's terminal emulator to decide whether or not to react to those codes.

For example, macOS's Terminal.app sees these escape codes and automatically resizes itself before playback starts. Here's what it looks like: https://www.loom.com/share/3ad4078e6ef841ecaad3c07afb5fb1c9?sid=bfea93e1-186b-4d3a-b2a6-3aa4e95e0a73

iTerm2 behaves differently. It has a dedicated setting for _Disable session-initiated window resizing_, but I was unable to get it to work even with this setting unchecked:
 
<img width="589" alt="image" src="https://github.com/gravitational/teleport/assets/7527103/42140bab-c357-41fd-8f37-55b8ee3cb5eb">

tl;dr: in order for console programs like vim to play back correctly, the terminal needs to be the correct size. Teleport is telling the terminal what size it should be, but some terminals choose not to listen to us.
